pkg/auth/auth_handler.go

```diff
@@ -232,6 +244,17 @@ func (a *AuthHandler) CreateAuthenticator() func(name, in string, authenticate s
 				}
 				return true, nil, common.NewInfraError(http.StatusUnauthorized, err)
 			}
+			//this code is part of the authorization process and should move to authz_handler
+			//after https://github.com/go-openapi/runtime/issues/158 is resolved
+			if payload, ok := p.(*ocm.AuthPayload); ok && !payload.IsAdmin {
+				if cluster_id := params.GetParam(r.Context(), params.ClusterId); cluster_id != "" {
+					if a.db != nil && !a.isClusterOwnedByUser(payload.Username, cluster_id) {
```

Comment:
do we need this 'db != nil' check? I think it must be defined if passed from main.

Reply:
Required to keep the tests independent of db and as a best practice